src/core/exception.d

```diff
@@ -433,9 +460,7 @@ deprecated void setAssertHandler( AssertHandler h ) @trusted nothrow @nogc
  */
 extern (C) void onAssertError( string file = __FILE__, size_t line = __LINE__ ) nothrow
 {
-    if( _assertHandler is null )
-        throw new AssertError( file, line );
-    _assertHandler( file, line, null);
+    onAssertErrorMsg(file, line, null);
```

Comment:
@WalterBright : Having two function seems unnecessary, as the case of `assert(cond)` could just be treated as `assert(cond, null)`. That's one less function between DMD and druntime, so less complication.
Also it's a common source of frustration that when debugging we need to break both on `_d_assert` and `_d_assert_msg`.